r/R/array.R

```diff
@@ -187,8 +187,27 @@ Array$create <- function(x, type = NULL) {
     }
     return(out)
   }
-  vec_to_Array(x, type)
+
+  if (is.null(type)) {
+    return(vec_to_Array(x, type))
+  }
+
+  # a type is given, this needs more care
+  tryCatch(
+    vec_to_Array(x, type),
+    error = function(cnd) {
+      attempt <- try(vec_to_Array(x, NULL)$cast(type), silent = TRUE)
+      abort(
+        c(conditionMessage(cnd),
+          i = if (!inherits(attempt, "try-error")) {
+            "You might want to try casting manually with `Array$create(...)$cast(...)`."
+          }
+        )
+      )
+    }
```

Comment:
Should we test this message (as well as the `try(vec_to_Array(x, NULL)$cast(type), silent = TRUE)` bit too)?

Reply:
Makes sense. In`test-Array.R`? @jonkeane

Reply:
Yeah, that looks like a good place for them

Reply:
Done